### PR TITLE
Add support for delay gate

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Needed to be updated to handle all possible units of Delay instruction duration, including dt, which requires knowing dt.

Aquire, Measure, and Drive channels should all get pulse.Delays in the output schedule.

### CODE REVIEW

File: qiskit/scheduler/methods/basic.py [hunk 1]
```diff
@@ -166,25 +166,44 @@ def get_measure_schedule() -> CircuitPulseDef:
...
         elif isinstance(inst, Delay):
-            sched = Schedule(name=inst.name)
-            for q in inst_qubits:
-                sched += pulse_inst.Delay(duration=inst.duration, channel=DriveChannel(q))  # OK?
-                sched += pulse_inst.Delay(duration=inst.duration, channel=MeasureChannel(q))  # OK?
+            if inst.unit == 'dt':
+                duration = inst.duration
+            else:
+                duration_s = convert_to_seconds(inst.duration, inst.unit)
+                duration = int(duration_s // schedule_config.dt)
```
Comment (itoko) [inline qiskit/scheduler/methods/basic.py:193]: I think rounding may be safer than rounding down, but it's a nit pick. I'll merge first, and improve later if necessary.

File: qiskit/compiler/sequence.py [hunk 2]
```diff
@@ -34,7 +34,8 @@
 def sequence(scheduled_circuits: Union[QuantumCircuit, List[QuantumCircuit]],
              backend: Optional[BaseBackend] = None,
              inst_map: Optional[InstructionScheduleMap] = None,
-             meas_map: Optional[List[List[int]]] = None) -> Union[Schedule, List[Schedule]]:
+             meas_map: Optional[List[List[int]]] = None,
+             dt: Optional[float] = None) -> Union[Schedule, List[Schedule]]:
```
Comment (itoko) [inline qiskit/compiler/sequence.py:38]: I'll merge this PR as it is to work MVP early. But I think `sequence` should not be aware of `dt`. Let's discuss this on slack.

**Comment by itoko** [comment]
@lcapelluto Could you resolve the conflicts so that I can merge this?

File: qiskit/scheduler/methods/basic.py (hunk 1 above)
Comment (lcapelluto) [reply qiskit/scheduler/methods/basic.py:193]: yes - we can also add a warning in another PR, too

File: qiskit/compiler/sequence.py (hunk 2 above)
Comment (lcapelluto) [reply qiskit/compiler/sequence.py:38]: Ah, actually, that's a really good point. I have to make changes to fix the conflicts anyway, so I'll fix this, too.

File: qiskit/scheduler/methods/lowering.py [hunk 3]
```diff
@@ -106,3 +116,14 @@ def get_measure_schedule() -> CircuitPulseDef:
         circ_pulse_defs.append(get_measure_schedule())
 
     return circ_pulse_defs
+
+def _convert_to_seconds(value: float, unit: str) -> float:
+    """Convert input value to seconds by applying unit to value."""
+    prefixes = {'p': 1e-12, 'n': 1e-9, 'u': 1e-6, 'µ': 1e-6,
+                'm': 1e-3, 'k': 1e3, 'M': 1e6, 'G': 1e9, 's': 1}
```
Comment (ajavadia) [inline qiskit/scheduler/methods/lowering.py:123]: can you make this part consistent with https://github.com/Qiskit/qiskit-terra/pull/4480- it was the cause of floating point issues

File: qiskit/scheduler/methods/lowering.py [hunk 4]
```diff
@@ -106,3 +116,14 @@ def get_measure_schedule() -> CircuitPulseDef:
         circ_pulse_defs.append(get_measure_schedule())
 
     return circ_pulse_defs
+
+def _convert_to_seconds(value: float, unit: str) -> float:
+    """Convert input value to seconds by applying unit to value."""
+    prefixes = {'p': 1e-12, 'n': 1e-9, 'u': 1e-6, 'µ': 1e-6,
```
Comment (chriseclectic) [inline qiskit/scheduler/methods/lowering.py:122]: If this is to be general unit dic i think 's' needs to be the empty string, and for the seconds dict we add s to all of them ie `'ns', 'ms', 's' etc,

**Comment by lcapelluto** [comment]
Thanks for replies @chriseclectic and @ajavadia . Maybe we can move the need to convert duration into the delay instruction?

**Comment by itoko** [comment]
Let's improve the issues on time conversion in the following PRs. I'll merge this as is for now.

**Pull request merged**